lib/manageiq/appliance_console/message_configuration.rb

```diff
@@ -177,6 +181,22 @@ def file_contains?(path, content)
         true
       end
 
+      def is_hostname?(host)
+        require "resolv"
+
+        return !(host =~ IP_REGEXP)
+      end
+
+      def hostname_fqdn?(host)
+        require "socket"
+
+        fqdn = Addrinfo.getaddrinfo(Socket.gethostname, nil).first.getnameinfo.first
```

Comment:
Not sure if its better to use `hostname -f` (with back ticks) here?

Reply:
What is the ` Addrinfo.getaddrinfo` doing for us?  Can you just use what `Socket.gethostname` returns?

Reply:
`Addrinfo.getaddrinfo` provides more info such as the FQDN whereas `Socket.gethostname` will provide the hostname but not necessarily the FQDN. I've tested this on an AWS appliance i.e.
```
irb(main):002:0> Addrinfo.getaddrinfo(Socket.gethostname, nil).first.getnameinfo.first
=> "<host>.us-west-1.compute.internal"
irb(main):003:0> Socket.gethostname
=> "<host>"
```

Reply:
Are we guaranteed to have the `addrinfo` we want "first" in the array and the `nameinfo` we want to also be first?

Googling around, I found my buddy from the old sql server adapter days, metaskills opened https://github.com/newrelic/newrelic-ruby-agent/issues number 697.  The newrelic ruby agent now defaults to trying `hostname -f` first, then `hostname`, and then heroku dyno name, and finally `gethostname`.  See: https://github.com/newrelic/newrelic-ruby-agent/pull/1015/files

And the current code in main:
https://github.com/newrelic/newrelic-ruby-agent/blob/0824b224b9c1be974d0b47ae9422f2f996046218/lib/new_relic/agent/hostname.rb#L11-L38

Reply:
This is a good suggestion. Thoughts on something like this (first time using awesome_spawn)?
```suggestion
        begin
          fqdn = AwesomeSpawn.run!("hostname", :params => ["-f"])
        rescue AwesomeSpawn::CommandResultError
          require "socket"

          fqdn = Addrinfo.getaddrinfo(Socket.gethostname, nil).first.getnameinfo.first
        end

```

Reply:
I think we'll be able to guarantee if we'll have `hostname` binary installed on appliances.  I guess my question is can we just assume `hostname` or do we need the fallback?  

If we need the fallback, do we need to worry about the order of the arrays (assumption on the first element being the correct one).  I recall some ruby stdlib for ip addresses would sometimes return loopback first instead of your publicly visible ip address so I wasn't sure if we have similar problems if there are multiple results.

Reply:
Before we do anything I'd like to understand how we got different values in the messaging yml and the cert.